aten/src/ATen/function_wrapper.py

```diff
@@ -596,7 +638,7 @@ def device_guard(option, dispatch_options, dispatch_tensor):
     # For factory methods the `DeviceGuard` is already in the template.
     if option.get('device_guard', True):
         if dispatch_options:
-            return 'const DeviceGuard device_guard({}.device());'.format(dispatch_options['name'])
+            return 'const OptionalDeviceGuard device_guard(device); '
```

Comment:
I'm not sure I follow this dispatch_options stuff -- this is for getting the DeviceGuard correct, but it's dependent on the dispatch.  And since the dispatch is now moved to BackendSelect, shouldn't this just move there?